File: matlab/rbfard2VardistPsi0Gradient.m
function [gKern, gVarmeans, gVarcovars] = rbfard2VardistPsi0Gradient(rbfard2Kern, vardist, covGrad)
% GGWHITEXGAUSSIANWHITEKERNGRADIENT Compute gradient between the GG white
%                                   and GAUSSIAN white kernels.
% FORMAT
% DESC computes the
%	gradient of an objective function with respect to cross kernel terms
%	between GG white and GAUSSIAN white kernels for the multiple output kernel.
% RETURN g1 : gradient of objective function with respect to kernel
%	   parameters of GG white kernel.
% RETURN g2 : gradient of objective function with respect to kernel
%	   parameters of GAUSSIAN white kernel.
% ARG ggwhitekern : the kernel structure associated with the GG white kernel.
% ARG gaussianwhiteKern :  the kernel structure associated with the GAUSSIAN white kernel.
% ARG x : inputs for which kernel is to be computed.
%
% FORMAT
% DESC  computes
%	the gradient of an objective function with respect to cross kernel
%	terms between GG white and GAUSSIAN white kernels for the multiple output kernel.
% RETURN g1 : gradient of objective function with respect to kernel
%	   parameters of GG white kernel.
% RETURN g2 : gradient of objective function with respect to kernel
%	   parameters of GAUSSIAN white kernel.
% ARG ggwhiteKern : the kernel structure associated with the GG white kernel.
% ARG gaussianwhiteKern : the kernel structure associated with the GAUSSIAN white kernel.
% ARG x1 : row inputs for which kernel is to be computed.
% ARG x2 : column inputs for which kernel is to be computed.
%
% SEEALSO :
% COPYRIGHT : Michalis K. Titsias
%

gKern = zeros(1,rbfard2Kern.nParams); 
gKern(1) = covGrad*vardist.numData;
 
gVarmeans = zeros(1,prod(size(vardist.means))); 
gVarcovars = zeros(1,prod(size(vardist.means))); 

File: R/vargplvm/R/rbfard2VardistPsi0Gradient.R
rbfard2VardistPsi0Gradient <-
function (rbfard2Kern, vardist, covGrad)
{
  # % RBFARD2VARDISTPSI0GRADIENT Description
  
  # % VARGPLVM
  gKern <- matrix(0, 1, rbfard2Kern$nParams)  
  gKern[1] <- covGrad*vardist$numData 
  
  gVarmeans <- matrix(0, 1, prod(dim(vardist$means)))  
  gVarcovars <- matrix(0, 1, prod(dim(vardist$means)))  
  return(list(gKern = gKern, gVarmeans = gVarmeans, gVarcovars = gVarcovars))
}
